src/Map.js

```diff
@@ -66,6 +66,17 @@ export class Map extends KeyedCollection {
     });
   }
 
+  static fromOwnEntries(source) {
+    return Map(
+      Reflect.ownKeys(source).map(key => [
+        key,
+        typeof source[key] === 'object'
+          ? Map.fromOwnEntries(source[key])
```

Comment:
Yeah I don't think this recursion is something we want to encourage - that would result in a deep exploration of a data structure which could be very slow, that might be surprising to someone who thought this worked like the other constructors which are all shallow - if an inner value is intended to also be a map, then explicitly calling this in user code will both make that use case clear, avoid unnecessary recursion and make ts/flow more accurate.

If I understand correctly, this function should just become:

```
return Map(Reflect.ownKeys(source).map(key => [key, source[key]]))
```

Given the simplicity of this function, how it is likely to be used less often than the standard constructor, how tree-shaking often does not yet remove static methods, and your pointing out that your use case may desire recursion to be more ergonomic, do you think it should live in the core library directly? I'm thinking that this could probably be either a single function in someone's own codebase or a micro library on npm (`immutable-map-from-own-entries`)?

Reply:
thanks again. i'm closing this request now.